### DISCUSSION TITLE
Correct JSON format for the JSON_LISTING_FEED product upload SP-API

### DISCUSSION THREAD

**Discussion by richd4** [amzn/selling-partner-api-models#3624] [General]
{
    "header": {
      "sellerId": "XXXXXXXXXXX",
      "version": "2.0",
      "issueLocale": "en_US"
    },
    "messages": [
      {
        "messageId": 1,
        "sku": "TESTSKU",
        "operationType": "UPDATE",
        "productType": "LUGGAGE",
        "requirements": "LISTING",
        "attributes": {
          "item_name": [
            {
              "value": "TEST Title",
              "language_tag": "en_US",
              "marketplace_id": "ATVPDKIKX0DER"
            }
          ],
          "fulfillment_availability": [
            {
              "fulfillment_channel_code": "AMAZON_NA",
              "quantity": 10
            }
          ],
          "purchasable_offer": [
            {
              "currency": "US",
              "our_price": [
                {
                  "schedule": [
                    {
                      "value_with_tax": 285.93
                    }
                  ]
                }
              ],
              "marketplace_id": "ATVPDKIKX0DER"
            }
          ]
        }
      }
    ]
  }

This is currently how my JSON is formatted for the JSON_LISTING_FEED upload channel and whenever I go to upload the test product I can see it processing in seller central, and once it is done the report says I am missing a bunch of attributes for the product upload. If anyone could give me any help on what else I may need to add in the JSON for it to be recognized as a valid product please let me know.

**Comment by rugved1991** [comment]
Hi @richd4,

You can check out this Listings workflow guide which talks about creating an offer_only product. You will need to call the Product Definitions API to understand the requirements for creating a listing and then provide all the required attributes to create a listing successfully.
https://developer-docs.amazon.com/sp-api/docs/building-listings-management-workflows-guide#list-an-offer-for-an-item-that-already-exists-in-the-amazon-catalog

Best,
Rugved
Solutions Architect, SP API
